# Examen Parcial 3.

Simulación.

**Grupo:** 001.

**Prof.** Lic. Blance Elida Serna Rodríguez.

Alma Cecilia Villarreal Duarte.

## Ejercicio 1.

**a)** Sea $X_{i}\sim Exp(1)$  $\forall$  i=1,2,3,4,5 

$\Rightarrow \theta=P(\sum_{i=1}^{5}i x_{i}\geq21.6)$

La suma se puede genera con una ciclo donde se sume cada i por el log(runif(1,5)).

Donde:

$\theta=I(\sum_{i=1}^{5}-i ln(U_{i})\geq21.6), siendo I, la integral de la función$

In [27]:
set.seed(1234)
Estimadortetha <- function(n, valor){
    sumatoria <- vector()
    j <- c(1:5)
    for(i in 1:n){
        u <- runif(5)
        sumatoria[i] <- sum(-(j*log(u))) >= valor
    }
    return(sumatoria)
}

Resulttetha <- Estimadortetha(10000, 21.6)
mean(Resulttetha)
var(Resulttetha)

[1] 0.1687

[1] 0.1402543

**b)** Usando las variables antitéticas, tenemos que el estimador es:


$I(\sum_{i=1}^{5}-i ln(U_{i})\geq21.6)+I(\sum_{i=1}^{5}-i ln(1-U_{i})\geq21.6)$

Siendo I, la integral de la función

In [28]:
set.seed(1234)
EstimadorPANT <- function(n){
    sumatoria1 <- vector()
    sumatoria2 <- vector()
    j <- c(1,2,3,4,5)
    for(i in 1:n){
        u <- runif(5)
        sumatoria1[i] <- sum(-(j*log(u))) >= 21.6
        sumatoria2[i] <- sum(-(j*log((1-u)))) >= 21.6
    }
    sumT <- (sumatoria1 + sumatoria2)/2
    return(sumT)
}

ResultPANT <- EstimadorPANT(10000)
mean(ResultPANT)
var(ResultPANT)

[1] 0.1664

[1] 0.05731677

**c)** Los resultados de las varianzas:

*estimador1=0.140254335433543 & estimador 2=0.0573167716771677

Donde un estimador es más eficiente o más preciso que otro estimador, si la varianza del primero es menor que la del segundo, es decir, $var(\theta_{1})<var(\theta_{2})$

In [34]:
print(var(Resulttetha)<var(ResultPANT))
eficiencia <- ((var(Resulttetha)/var(ResultPANT)))
eficiencia
Por_Error<-100*(var(Resulttetha) -var(ResultPANT))/var(Resulttetha)
Por_Error

[1] FALSE


[1] 2.447003

[1] 59.13369

El estimador de las variables antiteticas tiene una varianza menor, por lo que es un **estimador2 más eficiente que el estimador 1.**

Teniendo que la $Eficiencia=\frac{\theta_{2}}{\theta_{1}}=\frac{0.140254335433543}{0.0573167716771677}$

**Eficiencia=2.44700340457964**

Por lo que $\theta_{2}$ es 2.4470 veces más eficiente que $\theta_{1}$

Midiendo el porcentaje de reducción de error, tenemos que:

%Reducción de error=100*(0.140254335433543-0.0573167716771677)/0.140254335433543

**%Reducción de error=59.1336898784665%**

## Ejercicio 2.

**Inciso a**

**Aproximación de theta**

In [90]:
set.seed(1234)
n <- 10000
n1 <- rnorm(n,1,sqrt(1))
n2 <- rnorm(n,1,sqrt(1))
exp <- exp(n1*n2)
mean(exp) 
var(exp)

[1] 99.82906

[1] 6262157

**Inciso b**

**Forma 1**

**Aproximación con variable de control**

In [91]:
reg <- lm(exp ~ n1+n2)$coef
reg

(Intercept)          n1          n2 
  -318.6115    203.5712    215.2921

In [92]:
exp2<-reg[1]+reg[2]+reg[3]
exp2

(Intercept) 
   100.2519

In [93]:
exp2 <- exp - reg[2]*(n1-1)- reg[3]*(n2-1)
mean(exp2)
var(exp2)

[1] 100.2519

[1] 6173899

**Estimación del porcentaje de reducción en la varianza:**

In [98]:
por_var<-100*(var(exp)-var(exp2))/var(exp)
por_var

[1] 1.409382

**Forma 2**

In [99]:
Cov<-(exp-mean(exp))*(n1-mean(exp))
Cov<-sum(Cov)/(n-1)
Cov
Cov22<-mean(exp*n1)-(mean(exp)*mean(n1))
Cov22

[1] 200.8254

[1] 200.8053

In [101]:
VarCont<-var(exp)-(Cov^2/var(n1))
VarCont

[1] 6220801

In [102]:
Corr2<-Cov^2/(var(exp)*var(n1))
Corr2*100

[1] 0.6604093

In [ ]:
Mejora del 0.6604%